pilot/pkg/model/push_context.go

```diff
@@ -54,10 +54,15 @@ type PushContext struct {
 	privateVirtualServicesByNamespace map[string][]Config
 	publicVirtualServices             []Config
 
-	privateDestRuleHostsByNamespace  map[string][]Hostname
-	privateDestRuleByHostByNamespace map[string]map[Hostname]*combinedDestinationRule
-	publicDestRuleHosts              []Hostname
-	publicDestRuleByHost             map[Hostname]*combinedDestinationRule
+	// destination rules are of three types:
+	//  namespaceLocalDestRules: all public/private dest rules pertaining to a service defined in a given namespace
+	//  namespaceExportedDestRules: all public dest rules pertaining to a service defined in a namespace
+	//  allExportedDestRules: all (public) dest rules across all namespaces
+	// We need the allExportedDestRules in addition to namespaceExportedDestRules because we select
+	// the dest rule based on the most specific host match, and not just any destination rule
+	namespaceLocalDestRules    map[string]*processedDestRules
```

Comment:
Please add a comment on what is the key ( general comment, we miss this in a lot of places)